spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1966,8 +1969,8 @@ def _recover_from_error(self, exception, exc_info, executable_targets):
 
         # If there are any cores that are not in RTE, extract data from them
         if (len(non_rte_cores) > 0 and
-                self._executable_start_type ==
-                ExecutableStartType.USES_SIMULATION_INTERFACE):
+                ExecutableType.USES_SIMULATION_INTERFACE in
+                self._executable_types):
```

Comment:
It is worrying that nothing is below this change - you only want to extract data from those cores that are USES_SIMULATION_INTERFACE, otherwise the whole thing will grind to a halt.

Reply:
the extractor only extracts from cores using the simulation interface. the start type object has a map to cores. so this is safe to do

Reply:
OK, then this is fine :)